force-app/main/default/classes/SOQL.cls

```diff
@@ -288,54 +318,99 @@ public with sharing class SOQL {
-
-            // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
+         // Get the email from the Lead
+         // Check if the email is not null
+        // Add the email domain to the set
+        for(Lead listLeads : getLeads){
+            String emailDomain = listLeads.Email.substringAfter('@');
+            if(emailDomain != null && emailDomain != ''){
+                getDomains.add('www.' + emailDomain);
+            }
+        }
         // Query the Accounts
-        return null; // Replace null with the variable you used to store the result
+        List<Account> getAccounts = [SELECT Id, Name, Website FROM Account WHERE Website IN :getDomains];
+        return getAccounts; // Replace null with the variable you used to store the result
     }
 
-    /**
+        /**
      * Question 23
-     * Clones and Inserts Opportunity Line Items from an original Opportunity to a list of cloned Opportunities.
-     * Adds a randomized number between 1 to 3 to the quantity of each line item.
-     * Example: cloneLineItems('001XXXXXXXXXXXX', [list of opportunity ids]) should clone the line items from the Opportunity with the Id to each Opportunity in the list
-     * @param Original Opportunity Id
-     * @param List of cloned Opportunity Ids
-     */
-    public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-        // Query the original Opportunity Line Items
-
-        // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
-        // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
+/**
+ * Clones and Inserts Opportunity Line Items from an original Opportunity to a list of cloned Opportunities.
+ * Adds a randomized number between 1 to 3 to the quantity of each line item.
+ * 
+ * Example: cloneLineItems('001XXXXXXXXXXXX', [list of opportunity ids]) should clone the line items 
+ * from the Opportunity with the Id to each Opportunity in the list.
+ * 
+ * @param origOppId The Id of the original Opportunity.
+ * @param cloneOppIds The list of cloned Opportunity Ids.
+ */
+public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
+    // Query the original Opportunity Line Items
+    List<OpportunityLineItem> originalLineItems = [
+        SELECT Product2Id, Quantity, UnitPrice
+        FROM OpportunityLineItem
+        WHERE OpportunityId = :origOppId
+    ];
+
+    // Query the original Opportunity to get its Pricebook
+    Opportunity origOpp = [SELECT Pricebook2Id FROM Opportunity WHERE Id = :origOppId LIMIT 1];
+
+    // Ensure the cloned Opportunities have the same Pricebook2Id as the original Opportunity
+    List<Opportunity> clonedOpportunities = [
+        SELECT Id, Pricebook2Id
+        FROM Opportunity
+        WHERE Id IN :cloneOppIds
```

Comment:
Try to remove two of these queries and leave one, then try to use the clone() method